rita_common/src/rita_loop/write_to_disk.rs

```diff
@@ -0,0 +1,151 @@
+
+    match old_settings.clone() {
+        SettingsOnDisk::RitaClientSettings(old_settings_client) => {
+            match old_settings_client.network.device.clone() {
+                Some(val) => router_storage_small = is_router_storage_small(&val),
+                None => router_storage_small = false,
+            };
+            saving_to_disk_frequency = Duration::from_secs(old_settings_client.save_interval);
+        }
+        SettingsOnDisk::RitaExitSettingsStruct(old_settings_exit) => {
+            match old_settings_exit.network.device.clone() {
+                Some(val) => router_storage_small = is_router_storage_small(&val),
+                None => router_storage_small = false,
+            };
+            saving_to_disk_frequency = Duration::from_secs(old_settings_exit.save_interval);
+        }
+    }
+
+    thread::spawn(move || loop {
+        let start = Instant::now();
+
+        if start.elapsed() < saving_to_disk_frequency {
+            thread::sleep(saving_to_disk_frequency - start.elapsed());
+        }
+
+        //settings
+        match old_settings.clone() {
+            SettingsOnDisk::RitaClientSettings(old_settings_client) => {
+                let new_settings = get_rita_client();
+
+                if old_settings_client != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    log::info!("Config/settings is being saved to");
+                }
+
+                old_settings = SettingsOnDisk::RitaClientSettings(new_settings);
+            }
+            SettingsOnDisk::RitaExitSettingsStruct(old_settings_exit) => {
+                let new_settings = get_rita_exit();
+                saving_to_disk_frequency = Duration::from_secs(new_settings.save_interval);
+
+                if old_settings_exit != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    log::info!("Config/settings is being saved to");
+                }
+
+                old_settings = SettingsOnDisk::RitaExitSettingsStruct(new_settings);
+            }
+        }
+
+        // debt keeper
+        if !router_storage_small {
+            let runner = AsyncSystem::new();
+            runner.block_on(async move {
+                if let Err(e) = send_debt_update() {
+                    warn!("Debt keeper update failed! {:?}", e);
+                }
+                info!("Config/settings is being saved to");
+            });
+        }
+
+        //usage tracker is invoked via trafficwatch/watch() and the block runner
+        let babel_port = settings::get_rita_common().network.babel_port;
+        trace!("Common tick!");
+
+        let res = tm_get_neighbors();
+        trace!("Currently open tunnels: {:?}", res);
```

Comment:
this can all be removed you're not running usage tracker here? You should really only have to call it's save function here you can leave the setup work and normal calling in the fast loop.